chain/manager.go

```diff
@@ -1131,6 +1163,8 @@ func (m *Manager) updateV2TransactionProofs(txns []types.V2Transaction, from, to
 			return nil, fmt.Errorf("missing applied block at index %v", index)
 		} else if bs == nil {
 			return nil, fmt.Errorf("missing applied block supplement at index %v", index)
+		} else if err := m.overwriteExpirations(b, bs); err != nil {
```

Comment:
I wonder if we could move this into `blockAndChild` and `blockAndParent`? That seems to catch all spots where we need to call `overwriteExpirations`?